bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -207,10 +209,17 @@ public SingleDirectoryDbLedgerStorage(ServerConfiguration conf, LedgerManager le
         gcThread = new GarbageCollectorThread(conf,
                 ledgerManager, ledgerDirsManager, this, entryLogger, ledgerIndexDirStatsLogger);
 
+        // Because the writeCache and the writeCacheBeingFlushed are swapped during flushes, it is
+        // safer to acquire the underlying cache size and count references and then use those for metrics.
+        final AtomicLong finalWriteCacheSize = writeCache.cacheSize;
+        final AtomicLong finalWriteCacheBeingFlushedSize = writeCacheBeingFlushed.cacheSize;
+        final LongAdder finalWriteCacheCounter = writeCache.cacheCount;
+        final LongAdder finalWriteCacheBeingFlushedCounter = writeCacheBeingFlushed.cacheCount;
+
```

Comment:
I don't think this is right.
Lambdas in the DbLedgerStorageStats() constructor used to provide a way to access to metrics of current writeCache/.. instance.
With this change it will provide metrics for the first used instance, one before the very first swap, so the stats will stop changing after the cache swaps couple of times.

Reply:
Notice that we're aggregating the `writeCache` and `writeCacheBeingFlushed` size and count. Given that the swap is just to replace references for `writeCache` and `writeCacheBeingFlushed`, I am not sure I understand what is wrong here. In this PR, we grab the references once and remove the possibility of `writeCache.size() + writeCacheBeingFlushed.size()` double counting the same value from `writeCache` in the event the caches are swapped after `writeCache.size()` and before `writeCacheBeingFlushed.size()`